usaspending_api/common/query_with_filters.py

```diff
@@ -930,10 +943,38 @@ class QueryWithFilters:
 
     unsupported_filters = ["legal_entities"]
 
-    @classmethod
-    def _generate_elasticsearch_query(cls, filters: dict, query_type: _QueryType, **options) -> ES_Q:
-        nested_path = options.pop("nested_path", "")
+    def __init__(self, query_type: QueryType):
+        self.query_type = query_type
+        time_period_obj = None
+        if self.query_type == QueryType.ACCOUNTS:
+            self.default_options = {"nested_path": "financial_accounts_by_award"}
+        elif self.query_type == QueryType.TRANSACTIONS:
+            time_period_obj = TransactionSearchTimePeriod(
+                default_end_date=settings.API_MAX_DATE, default_start_date=settings.API_MIN_DATE
+            )
+        elif self.query_type == QueryType.AWARDS:
+            time_period_obj = AwardSearchTimePeriod(
+                default_end_date=settings.API_MAX_DATE, default_start_date=settings.API_MIN_DATE
+            )
+        elif self.query_type == QueryType.SUBAWARDS:
+            time_period_obj = SubawardSearchTimePeriod(
+                default_end_date=settings.API_MAX_DATE, default_start_date=settings.API_MIN_DATE
+            )
+            self.default_options = dict()
 
+        if time_period_obj is not None and (self.query_type == QueryType.AWARDS or self.query_type == QueryType.TRANSACTIONS):
+            new_awards_only_decorator = NewAwardsOnlyTimePeriod(
+                time_period_obj=time_period_obj, query_type=self.query_type
+            )
+            self.default_options = {"time_period_obj": new_awards_only_decorator}
+        if self.query_type == QueryType.ACCOUNTS:
+            self.default_options = (
+                {"nested_path": "financial_accounts_by_award"}
+            )
+
+    def generate_elasticsearch_query(self, filters: dict, **options) -> ES_Q:
+        options = options if len(options) > 0 else self.default_options
```

Comment:
I am curious, what problems did you run into with your previous change that resulted in you going to this? Referencing this: https://github.com/fedspendingtransparency/usaspending-api/pull/4289/commits/30107e164cd2af3ab97351320bf4f3bd9215aa6d#diff-45ff8631ebb70d95ce8525bfa64ae4d79edb9d19b13088cafbdd0e996078b11fL976

Reply:
I believe what you meant to use in your previous commit was `options = {**self.default_options, **options}`, that way any user provided options will override the default options that are spread across the dictionary.